From: Takashi Iwai <tiwai@suse.de>
Subject: kABI workaround for ppc KVM VTB fix
Patch-mainline: Never, SLE12-SP2 kabi fix
References: bsc#981367

Keep the kABI compatibility for the patch
patches.arch/KVM-PPC-Book3S-Treat-VTB-as-a-per-subcore-register-not-per-thread

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 arch/powerpc/include/asm/kvm_book3s.h |    4 +++-
 arch/powerpc/include/asm/kvm_host.h   |    5 ++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

--- a/arch/powerpc/include/asm/kvm_host.h
+++ b/arch/powerpc/include/asm/kvm_host.h
@@ -297,8 +297,10 @@ struct kvmppc_vcore {
 	u32 arch_compat;
 	ulong pcr;
 	ulong dpdes;		/* doorbell state (POWER8) */
-	ulong vtb;		/* virtual timebase */
 	ulong conferring_threads;
+#ifndef __GENKSYMS__
+	ulong vtb;		/* virtual timebase */
+#endif
 };
 
 #define VCORE_ENTRY_MAP(vc)	((vc)->entry_exit_map & 0xff)
@@ -474,6 +476,7 @@ struct kvm_vcpu_arch {
 	ulong purr;
 	ulong spurr;
 	ulong ic;
+	ulong vtb;	/* just a placeholder for kABI compatibility */
 	ulong dscr;
 	ulong amr;
 	ulong uamor;
--- a/arch/powerpc/include/asm/kvm_book3s.h
+++ b/arch/powerpc/include/asm/kvm_book3s.h
@@ -83,7 +83,6 @@ struct kvmppc_vcpu_book3s {
 	u64 sdr1;
 	u64 hior;
 	u64 msr_mask;
-	u64 vtb;
 #ifdef CONFIG_PPC_BOOK3S_32
 	u32 vsid_pool[VSID_POOL_SIZE];
 	u32 vsid_next;
@@ -105,6 +104,9 @@ struct kvmppc_vcpu_book3s {
 #endif
 	int hpte_cache_count;
 	spinlock_t mmu_lock;
+#ifndef __GENKSYMS__
+	u64 vtb;
+#endif
 };
 
 #define VSID_REAL	0x07ffffffffc00000ULL
